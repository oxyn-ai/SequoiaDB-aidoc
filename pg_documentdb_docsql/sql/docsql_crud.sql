CREATE EXTENSION IF NOT EXISTS pg_documentdb;
CREATE EXTENSION IF NOT EXISTS pg_documentdb_docsql;

SELECT documentdb_docsql.create_database('docdb1');
SELECT documentdb_docsql.create_collection('docdb1','patient');

INSERT INTO docsql_docdb1.patient(document)
VALUES ('{ "patient_id": "P001", "name": "Alice", "age": 30 }');

SELECT (document ->> 'patient_id') AS pid, (document ->> 'name') AS name
FROM docsql_docdb1.patient
ORDER BY pid;
