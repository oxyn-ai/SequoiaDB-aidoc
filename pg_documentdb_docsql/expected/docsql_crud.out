CREATE EXTENSION IF NOT EXISTS pg_documentdb;
CREATE EXTENSION IF NOT EXISTS pg_documentdb_docsql;
SELECT documentdb_docsql.create_database('docdb1');
 create_database 
-----------------
 
(1 row)

SELECT documentdb_docsql.create_collection('docdb1','patient');
 create_collection 
-------------------
 
(1 row)

INSERT INTO docsql_docdb1.patient(document)
VALUES ('{ "patient_id": "P001", "name": "Alice", "age": 30 }');
INSERT 0 1
SELECT (document ->> 'patient_id') AS pid, (document ->> 'name') AS name
FROM docsql_docdb1.patient
ORDER BY pid;
 pid | name 
-----+------
 P001 | Alice
(1 row)
